sdk/helper/keysutil/policy.go

```diff
@@ -1348,14 +1388,19 @@ func (p *Policy) VerifySignature(context, input []byte, hashAlgorithm HashType,
 
 		switch sigAlgorithm {
 		case "pss":
-			err = rsa.VerifyPSS(&key.PublicKey, algo, input, sigBytes, nil)
+			// At the time of writing, this constant has the lowest sentinel value:
+			// https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/crypto/rsa/pss.go;l=247
+			if saltLength < rsa.PSSSaltLengthEqualsHash {
+				return false, errutil.UserError{Err: fmt.Sprintf("requested salt length %d is invalid", saltLength)}
+			}
+			err = rsa.VerifyPSS(&key.PublicKey, algo, input, sigBytes, &rsa.PSSOptions{SaltLength: saltLength})
 		case "pkcs1v15":
 			err = rsa.VerifyPKCS1v15(&key.PublicKey, algo, input, sigBytes)
 		default:
 			return false, errutil.InternalError{Err: fmt.Sprintf("unsupported rsa signature algorithm %s", sigAlgorithm)}
 		}
 
-		return err == nil, nil
+		return err == nil, err
```

Comment:
Ah good catch! This isn't exactly right though, as the verify function is trying to distinguish between a verification failure and an actual error due to input/solar flares, so we wouldn't want to report the `rsa.ErrVerification` back to the caller as an error. So possible this would work?

```go
if errors.Is(err, rsa.ErrVerification) {
    return false, nil
}
return err != nil, err
```

Reply:
I'm probably going to revert this for two reasons: (1) this PR is not about this issue, and (2) it may be responsible for the test failures I haven't seen locally yet.

Reply:
Sounds good, I'll create another PR for this personally then and mention you in the PR description if you don't mind.

Reply:
Filed: https://github.com/hashicorp/vault/pull/16695